src/ngraph/runtime/backend.hpp

```diff
@@ -117,4 +119,23 @@ class ngraph::runtime::Backend
     // \param op_name is the name of the backend specific op
     // \returns a shared pointer to the op if found, else nullptr
     virtual std::shared_ptr<ngraph::Node> get_backend_op(const std::string& op_name, ...);
+
+    /// \brief Returns memory allocator used by backend for host allocations
+    virtual Allocator* get_host_memory_allocator()
```

Comment:
we need both, since we need to provide a way for framework to pass its allocator or the device can pass its allocator to the backend caller to allocated memory on the device(GPU/ Crest etc)

Reply:
ok, I see, so we should have both `m_host_allocator` and `m_device_allocator`, right?

Reply:
right :)